src/gui/mainwindow.cpp

```diff
@@ -24,16 +24,10 @@ MainWindow::MainWindow(NeovimConnector* c, QWidget* parent)
 
 void MainWindow::init(NeovimConnector *c)
 {
-	if (m_shell) {
-		m_shell->deleteLater();
-		m_stack.removeWidget(m_shell);
-	}
+	assert(!m_shell);
```

Comment:
This isn't functionally equivalent. The code before will gracefully handle `!m_shell`. This code will abort with an error.

I think this block is necessary for when the `ErrorWidget` displays on connection failure?

Reply:
I think the initialization code should just do that. Handling the case where `m_shell` and `m_nvim` is not properly uninitialized could hide some other bugs.

> I think this block is necessary for when the ErrorWidget displays on connection failure?

It doesn't seem to effect it. Like I said in my other comment, by the time the error widget is shown the init is called and if it didn't do what it's supposed to, retrying properly releases the previous resources and creates new ones.

Reply:
Please use the old logic here.

Let's not change things we don't have to during the refactor.